lib/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,28 @@
+import { useState } from "react";
+
+export const useLocalStorage = (key: string, defaultValue: unknown) => {
+  const [localStorageValue, setLocalStorageValue] = useState(() => {
+    try {
+      // Try and retrieve any previously recorded values
+      const value = localStorage.getItem(key);
```

Comment:
I think you could name this `currentValue`/`previousValue` and then just do away with the comments